components/brave_federated/task/federated_task_runner.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_FEDERATED_TASK_FEDERATED_TASK_RUNNER_H_
+#define BRAVE_COMPONENTS_BRAVE_FEDERATED_TASK_FEDERATED_TASK_RUNNER_H_
+
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "base/memory/raw_ptr.h"
+#include "brave/components/brave_federated/task/model.h"
+#include "brave/components/brave_federated/task/typing.h"
+#include "brave/components/brave_federated/util/linear_algebra_util.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_federated {
+
+class Model;
+class Task;
+using ModelWeights = std::tuple<Weights, float>;
+
+class FederatedTaskRunner final {
+ public:
+  FederatedTaskRunner(Task task, std::unique_ptr<Model> model);
+  ~FederatedTaskRunner();
+
+  absl::optional<TaskResult> Run();
+
+  void SetTrainingData(DataSet training_data);
```

Comment:
all of these should be `const &` https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs
"Non-optional input parameters should usually be values or const references"

Reply:
^^ applies everywhere